src/main/java/ch/njol/skript/util/NamespacedUtils.java

```diff
@@ -0,0 +1,124 @@
+package ch.njol.skript.util;
+
+import ch.njol.skript.Skript;
+import ch.njol.util.Pair;
+import com.google.common.collect.Sets;
+import org.apache.commons.lang.ArrayUtils;
+import org.bukkit.NamespacedKey;
+
+import java.util.Set;
+
+public class NamespacedUtils {
+
+	private static final Set<Character> LEGAL_NAMESPACE_CHARS = Sets.newHashSet(ArrayUtils.toObject("abcdefghijklmnopqrstuvwxyz0123456789._-/".toCharArray()));
+
+	/**
+	 * Gets a namespaced key. This method will try to get existing keys first, but if that fails
+	 * it will create the key in Skript's namespace.
+	 * @param key the unparsed key
+	 * @param usingMinecraft whether to have it as Skript or Minecraft namespace
+	 * @return the resulting NamespacedKey
+	 */
+	public static NamespacedKey getNamespacedKey(String key, Boolean usingMinecraft) {
```

Comment:
```suggestion
	public static NamespacedKey getNamespacedKey(String key, boolean usingMinecraft) {
```
usingMinecraft has no reason to be nullable